cpp/text_module/algorithm/text.cpp

```diff
@@ -0,0 +1,29 @@
+#include "text.hpp"
+
+void Text::Join(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const auto list{arguments[0].ValueList()};
+    const auto delimiter{arguments[1].ValueString()};
+
+    std::string result;
+    if (!list.Empty()) {
+      auto iterator = list.begin();
+      result += (*iterator).ValueString();
+
+      for (++iterator; iterator != list.end(); ++iterator) {
+        result += delimiter;
+        result += (*iterator).ValueString();
+      }
+    }
```

Comment:
Make use of early return. So:
if(list.Empty()){
  auto record = record_factory.NewRecord();
  record.Insert(std::string(kResultJoin).c_str(), result);
  return;
}
...